### ISSUE TITLE
InvalidCastException in UseExpressionBodyForConstructorsCodeFixProvider

### ISSUE THREAD

**Issue by mavasani** [dotnet/roslyn#16693]
**Version Used**: 
D15PreRel 26120.4

**Steps to Reproduce**:

1. Enable full solution analysis for C#
2. `git clone https://github.com/dotnet/roslyn-analyzers`
3. `msbuild BuildAndTest.proj /t:RestorePackages` - Verify restore succeeds.
4. `msbuild BuildAndTest.proj /m` - Verify build succeeds.
5. Open `src\Analyzers.sln`
6. Verify IDE0020 diagnostic (Use pattern matching) in `src\Microsoft.ApiDesignGuidelines.Analyzers\CSharp\CSharpTypesThatOwnDisposableFieldsShouldBeDisposable.cs` at line 31.
7. Ctrl + Dot, and attempt to apply code fix "Use block body for constructors" (PS: seems weird to have been offered this fix as I am in an if statement, not a constructor).

**Expected Behavior**:
Code fix works.

**Actual Behavior**:
```
System.InvalidCastException : Unable to cast object of type 'Microsoft.CodeAnalysis.CSharp.Syntax.IfStatementSyntax' to type 'Microsoft.CodeAnalysis.CSharp.Syntax.ConstructorDeclarationSyntax'.
   at Microsoft.CodeAnalysis.CSharp.UseExpressionBody.AbstractUseExpressionBodyCodeFixProvider`1.AddEdits(SyntaxEditor editor,Diagnostic diagnostic,OptionSet options,Boolean preferExpressionBody,CancellationToken cancellationToken)
   at Microsoft.CodeAnalysis.CSharp.UseExpressionBody.AbstractUseExpressionBodyCodeFixProvider`1.FixAllAsync(Document document,ImmutableArray`1 diagnostics,SyntaxEditor editor,CancellationToken cancellationToken)
   at Microsoft.CodeAnalysis.CodeFixes.SyntaxEditorBasedCodeFixProvider.<>c__DisplayClass2_0.<FixAllAsync>b__0(SyntaxEditor editor)
   at async Microsoft.CodeAnalysis.CodeFixes.SyntaxEditorBasedCodeFixProvider.FixAllWithEditorAsync(<Unknown Parameters>)
   at async Microsoft.CodeAnalysis.CodeActions.CodeAction.GetChangedSolutionAsync(<Unknown Parameters>)
   at async Microsoft.CodeAnalysis.CodeActions.CodeAction.ComputeOperationsAsync(<Unknown Parameters>)
   at async Microsoft.CodeAnalysis.CodeActions.CodeAction.ComputePreviewOperationsAsync(<Unknown Parameters>)
   at async Microsoft.CodeAnalysis.CodeActions.CodeAction.GetPreviewOperationsAsync(<Unknown Parameters>)
   at async Microsoft.CodeAnalysis.Editor.Implementation.Suggestions.SuggestedAction.GetPreviewResultAsync(<Unknown Parameters>)
   at async Microsoft.CodeAnalysis.Editor.Implementation.Suggestions.SuggestedActionWithNestedFlavors.<>c__DisplayClass11_0.<GetPreviewAsync>b__0(<Unknown Parameters>)
   at async Microsoft.CodeAnalysis.Extensions.IExtensionManagerExtensions.PerformFunctionAsync[T](<Unknown Parameters>)
   at Microsoft.VisualStudio.Telemetry.WindowsErrorReporting.WatsonReport.GetClrWatsonExceptionInfo(Exception exceptionObject)
```

**Comment by Pilchie** [comment]
Fixed by https://github.com/dotnet/roslyn/pull/16679

**Comment by Pilchie** [comment]
@CyrusNajmabadi Can you confirm this is fixed by #16679?

**Comment by CyrusNajmabadi** [comment]
Yes.  This was fixed.  thanks!
